assets/src/dashboard/app/views/editorSettings/publisherLogo/index.js

```diff
@@ -66,39 +71,157 @@ function PublisherLogoSettings({
   publisherLogos,
   uploadError,
 }) {
+  const { isRTL } = useConfig();
+
+  const containerRef = useRef();
+  const gridRef = useRef();
+  const itemRefs = useRef({});
+
+  const [activePublisherLogo, _setActivePublisherLogoId] = useState(null);
+  const [indexRemoved, setIndexRemoved] = useState(null);
+
+  const publisherLogosById = useMemo(() => publisherLogos.map(({ id }) => id), [
+    publisherLogos,
+  ]);
+
+  const [publisherLogoCount, setPublisherLogoCount] = useState(
+    publisherLogosById.length
+  );
+
+  const onRemoveLogoClick = useCallback(
+    (e, { publisherLogo, idx }) => {
+      e.preventDefault();
+
+      handleRemoveLogo(publisherLogo);
+      setIndexRemoved(idx);
+      setPublisherLogoCount(publisherLogosById.length);
+    },
+    [handleRemoveLogo, publisherLogosById.length]
+  );
+
+  const setActivePublisherLogoId = useCallback((id) => {
+    _setActivePublisherLogoId(id);
+  }, []);
+
+  // set active logo when first painting
+  useEffect(() => {
+    if (publisherLogos.length > 0 && !activePublisherLogo) {
+      setActivePublisherLogoId(publisherLogos?.[0].id);
+    }
+  }, [activePublisherLogo, publisherLogos, setActivePublisherLogoId]);
+
+  // Update publisher logo focus when logo is removed
+  useEffect(() => {
+    if (
+      Boolean(indexRemoved?.toString()) &&
+      publisherLogosById.length !== publisherLogoCount
```

Comment:
Then finally `publisherLogosById.length !== publisherLogoCount.current`.

We don't need to fire off a render to update the focus to the default because we are selecting it via the containerRef's firstChild.

Reply:
perfect, thank you!